### PR TITLE
Stop Rust from prepending underscore before '?' for win32

### LINKED ISSUE #819: Mangling hack doesn't work well with variable symbols
### Input C/C++ Header

```C++
struct Pref {
  static bool sX;
};
```

### Bindgen Invocation

```
$ bindgen test.hpp -- --target=i686-pc-win32
```

### Actual Results

```rust
extern "C" {
    #[link_name = "sX@Pref@@2_NA"]
    pub static mut Pref_sX: bool;
}
```

### Expected Results

```rust
extern "C" {
    #[link_name = "?sX@Pref@@2_NA"]
    pub static mut Pref_sX: bool;
}
```

### PR DESCRIPTION
**Pull request by upsuper**
This fixes #819.

It also includes tests for different platforms which are not supposed to be affected, so that we won't regress them in the future either.

The prefix `\x01` char is necessary for Win32. See also [msvc32_symbolify function in regen_atoms.py](https://github.com/servo/servo/blob/1b6d29e31996c87218352b825aa93e01909a6a24/components/style/gecko/regen_atoms.py#L35-L38).

### CODE REVIEW

File: src/ir/function.rs [hunk 1]
```diff
@@ -130,7 +130,14 @@ fn get_abi(cc: CXCallingConv) -> Abi {
 
 fn mangling_hack_if_needed(ctx: &BindgenContext, symbol: &mut String) {
     if ctx.needs_mangling_hack() {
-        symbol.remove(0);
+        match symbol.chars().next().unwrap() {
+            // Stripping leading underscore for all names on Darwin and
+            // C linkage functions on Win32.
+            '_' => { symbol.remove(0); }
```
Comment (emilio) [inline src/ir/function.rs]: nit: No need for braces, here and below.

Comment (upsuper) [inline src/ir/function.rs]: No, it needs. `String::remove` returns `char` and `String::insert` returns nothing.

**Pull request merged**